### PR TITLE
[REG2.055/2.063] Issue 9279 - Static array return value implicitly converted to immutable dynamic array

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=9279

Implement more escape analysis.

### CODE REVIEW

File: src/escape.c [hunk 1]
```diff
@@ -258,6 +251,12 @@ bool checkEscapeRef(Scope *sc, Expression *e, bool gag)
                 ::error(loc, format, s->toChars());
             result = true;
         }
+        void error(Loc loc, const char *format, Expression *e)
```
Comment (yebblies) [inline src/escape.c]: I'm a little concerned about adding yet another `error` function.  How much uglier would it be without it?

Comment (9rnsr) [inline src/escape.c]: I know someone said that we should not do both refactoring and regression fix in a PR.

Comment (9rnsr) [inline src/escape.c]: OK, I merged to `error` functions into one, by taking `RootObject *o`.

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Comment by WalterBright** [comment]
once the magicport issue is solved...

**Comment by 9rnsr** [comment]
@WalterBright Sorry, fixed ddmd issue.

**Comment by CyberShadow** [comment]
I think this PR broke building the docs:

```
../phobos/std/path.d(2362): Error: pure function 'std.path.absolutePath' cannot call impure function 'std.path.isAbsolute!string.isAbsolute'
../phobos/std/path.d(2362): Error: safe function 'std.path.absolutePath' cannot call system function 'std.path.isAbsolute!string.isAbsolute'
../phobos/std/path.d(2364): Error: pure function 'std.path.absolutePath' cannot call impure function 'std.path.isAbsolute!string.isAbsolute'
../phobos/std/path.d(2364): Error: safe function 'std.path.absolutePath' cannot call system function 'std.path.isAbsolute!string.isAbsolute'
../phobos/std/path.d(1545): Error: pure function 'std.path.buildNormalizedPath!char.buildNormalizedPath' cannot call impure function 'std.path.isAbsolute!(const(char)[]).isAbsolute'
../phobos/std/path.d(1545): Error: 'std.path.isAbsolute!(const(char)[]).isAbsolute' is not nothrow
../phobos/std/path.d(1512): Error: function 'std.path.buildNormalizedPath!char.buildNormalizedPath' is nothrow yet may throw
../phobos/std/datetime.d(28084): Error: template instance std.path.buildNormalizedPath!char error instantiating
make: *** [docs-prerelease.json] Error 1
```

**Comment by 9rnsr** [comment]
@CyberShadow The problem is introduced by the Phobos change: https://github.com/D-Programming-Language/phobos/pull/3381

**Comment by CyberShadow** [comment]
Thanks. (Strange, it was merged at 04:52 but my 05:00 build succeeded)

**Pull request merged**